astropy/io/fits/column.py

```diff
@@ -1429,6 +1429,12 @@ def dtype(self):
                 else:
                     dt = np.dtype((dt.base, dim))
 
+            # On Python 2, force the `name` to `str`
+            # because Numpy structured arrays can't handle unicode `name`
+            # See https://github.com/astropy/astropy/issues/5204
+            if PY2:  # pragma: py2
+                name = str(name)
```

Comment:
This will work only for strings with ascii-only characters. I was not sure if it make sense to support non-ascii characters, and looking at the FITS standard, it seems column names must have only ascii so it is fine.

> If the value of the TFORMn keyword specifies data type A, field n shall contain a character string of zero or more
members, composed of the restricted set of ASCII text characters.